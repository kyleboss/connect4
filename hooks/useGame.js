import { useContext } from "react";

import GameContext from "@/state/GameContext";

const useGame = () => useContext(GameContext);

export default useGame;
