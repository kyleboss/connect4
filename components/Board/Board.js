import styled from "styled-components";

import useGame from "@/hooks/useGame";
import Slot from "@/components/Slot";

const Container = styled.div`
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  gap: 10px;
  margin: 0 20px 50px 0;
`;

const Board = () => {
  const { board, makeMove, currentPlayer, userRole } = useGame();

  return (
    <Container>
      {board.map((row, rowIndex) =>
        row.map((cell, columnIndex) => (
          <Slot
            key={`${rowIndex}-${columnIndex}`}
            value={cell}
            onClick={() => makeMove(columnIndex)}
            disabled={currentPlayer !== userRole || cell}
          />
        ))
      )}
    </Container>
  );
};

export default Board;
