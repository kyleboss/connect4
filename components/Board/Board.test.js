import { render, fireEvent } from "@testing-library/react";

import useGame from "../../hooks/useGame";

import Board from "./Board";

jest.mock("../../hooks/useGame", () => ({
  __esModule: true,
  default: jest.fn(() => ({
    board: [[null, null, null, null, null, null, null]],
    makeMove: jest.fn(),
    currentPlayer: "Red",
    userRole: "Red",
  })),
}));

describe("Board", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Board />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of slots", () => {
    useGame.mockImplementation(() => ({
      board: [[null, null, null, null, null, null, null]],
      makeMove: jest.fn(),
      currentPlayer: "Red",
      userRole: "Red",
    }));
    const { getAllByRole } = render(<Board />);
    expect(getAllByRole("button").length).toBe(7);
  });

  it("calls makeMove when a slot is clicked", () => {
    const makeMoveMock = jest.fn();
    useGame.mockImplementation(() => ({
      board: [[null, null, null, null, null, null, null]],
      makeMove: makeMoveMock,
      currentPlayer: "Red",
      userRole: "Red",
    }));
    const { getAllByRole } = render(<Board />);
    const slots = getAllByRole("button");
    fireEvent.click(slots[0]);
    expect(makeMoveMock).toHaveBeenCalledWith(0);
  });
});
