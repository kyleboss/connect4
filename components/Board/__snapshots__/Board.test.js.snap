// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-e1fcfd44-0 dyIQsa"
  >
    <div
      class="sc-2b7f7c93-0 bONUyQ"
      role="button "
    />
    <div
      class="sc-2b7f7c93-0 bONUyQ"
      role="button "
    />
    <div
      class="sc-2b7f7c93-0 bONUyQ"
      role="button "
    />
    <div
      class="sc-2b7f7c93-0 bONUyQ"
      role="button "
    />
    <div
      class="sc-2b7f7c93-0 bONUyQ"
      role="button "
    />
    <div
      class="sc-2b7f7c93-0 bONUyQ"
      role="button "
    />
    <div
      class="sc-2b7f7c93-0 bONUyQ"
      role="button "
    />
  </div>
</DocumentFragment>
`;
